osu.Game/Screens/OnlinePlay/Playlists/AddPlaylistToCollectionButton.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using System.Threading.Tasks;
+using osu.Framework.Allocation;
+using osu.Framework.Extensions;
+using osu.Game.Collections;
+using osu.Game.Database;
+using osu.Game.Graphics;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Graphics.UserInterfaceV2;
+using osu.Game.Online.Rooms;
+using osu.Game.Overlays;
+using osu.Game.Overlays.Notifications;
+
+namespace osu.Game.Screens.OnlinePlay.Playlists
+{
+    public partial class AddPlaylistToCollectionButton : RoundedButton
+    {
+        private readonly Room room;
+
+        private LoadingLayer loading = null!;
+
+        [Resolved]
+        private RealmAccess realmAccess { get; set; } = null!;
+
+        [Resolved]
+        private BeatmapLookupCache beatmapLookupCache { get; set; } = null!;
+
+        [Resolved(canBeNull: true)]
+        private INotificationOverlay? notifications { get; set; }
+
+        public AddPlaylistToCollectionButton(Room room)
+        {
+            this.room = room;
+            Text = "Add Maps to Collection";
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours)
+        {
+            BackgroundColour = colours.Gray5;
+
+            Add(loading = new LoadingLayer(true, false));
+
+            Action = () =>
+            {
+                int[] ids = room.Playlist.Select(item => item.Beatmap.OnlineID).Where(onlineId => onlineId > 0).ToArray();
+
+                if (ids.Length == 0)
+                {
+                    notifications?.Post(new SimpleErrorNotification { Text = "Cannot add local beatmaps" });
+                    return;
+                }
+
+                Enabled.Value = false;
+                loading.Show();
+                beatmapLookupCache.GetBeatmapsAsync(ids).ContinueWith(task => Schedule(() =>
```

Comment:
I'd hope this isn't required. Why can't you just match using online IDs instead?

Reply:
I was basing it off of [`OnlinePlayBeatmapAvailabilityTracker`](https://github.com/ppy/osu/blob/0235928fdab73d1c7326297a9f9650201fc6c21b/osu.Game/Online/Rooms/OnlinePlayBeatmapAvailabilityTracker.cs#L75C17-L84C68) which is called in the Ready button in the playlist footer.

I can try changing it to see if it works. If I do, what should I use to lookup more information? Is it available in realm?

Reply:
I'm not sure if having the playlist contain non-downloaded beatmaps makes sense in the first place?

Although it might be preferred if the user clicks the button before having all beatmaps downloaded.

Maybe the button should say "Create playlist with x downloaded beatmaps"?

Reply:
Currently have it set up like this where both the beatmap count and whether the collection exists is shown in the button text. I also changed the wording so that it fits at all window sizes. Not sure on current wording and can't think of anything that wouldn't get truncated with smaller width screens.

Note: I also updated the text to be lower case like other buttons (not shown in video)

https://github.com/user-attachments/assets/351deecf-e61a-4a5c-84f5-b4b4d2d7129a